mapbox/dr-ui
# Fix bug in NumberedCodeSnippet

This PR is primarily fixing a bug in `NumberedCodeSnippet` where if the last line of the code snippet being used was included in a `copyRange`, the chunk would be marked as `live: false` when it should be `live: true`.

The only approach I could get to work was to loop through all lines _plus_ one more time to give it the chance to `endCurrentChunk` after the last line, but not push anything to the `currentChunk` the last time it loops through. I have a hunch @katydecorah might be able to come up with a more elegant solution, but this one seems to work!

In addition to this bug fix, the PR also:
- Adds test cases for when the `copyRange` includes the first line and the last line.
- Prevents showing Hide/Show lines if there are no line elements (this was happening at the beginning and end of code snippets).

## How to test

- Run `npm start`.
- Visually check all test cases for `NumberedCodeSnippet`.
    - The "First line" test case should display the first five lines of the sample code snippet and only include `Show hidden lines` at the bottom of the live chunk.
    - The "Last line" test case should display the last five lines of the sample code snippet and only include `Show hidden lines` at the top of the live chunk.

## QA checklist



- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `#.#.#`. (will be 91% until https://github.com/mapbox/dr-ui/pull/266 merges)
- [x] Create a PR in a site repo, copy the component, and test it. Push to staging and let the reviewer know they can also test the component there. (tested myself in /help 😬)

Open the test cases app locally on:


cc @chezzdev
